### ISSUE TITLE
stoi argument out of range

### ISSUE THREAD

**Issue by denelon** [microsoft/winget-cli#1549]
```
PS C:\Users\denelon> winget search 123 -n 853837271271781839344
An unexpected error occurred while executing the command:
stoi argument out of range
```

**Comment by pratikpc** [comment]
@denelon I believe the same

If you have custom exceptions, maybe the solution would be to

```cpp
try{
                intValue = std::stoi(str);
} catch(std::out_of:range&) {
                throw winget::exception("We cannot handle values that huge for count"); // If we have custom winget exceptions
}
if (intValue < 0)
                throw winget::exception("Count must always be positive");
```

And probably added unit tests to check for the same

**Comment by jedieaston** [comment]
This was resolved by #1564.
